gstaichi/program/launch_context_builder.cpp

```diff
@@ -224,8 +224,10 @@ void LaunchContextBuilder::set_arg_external_array_with_shape(
                  "External array cannot have > {max_num_indices} indices");
   array_ptrs[concatenate_vector<int>(
       arg_id, {TypeFactory::DATA_PTR_POS_IN_NDARRAY})] = (void *)ptr;
-  array_ptrs[concatenate_vector<int>(
-      arg_id, {TypeFactory::GRAD_PTR_POS_IN_NDARRAY})] = (void *)grad_ptr;
+  if (grad_ptr != 0) {
```

Comment:
why is this conditional being added? Was the absence of the conditional a bug before?

Reply:
No, just an optimisation. Not calling dealloc on nullptr.